syft/pkg/cataloger/java/graalvm_native_image_cataloger.go

```diff
@@ -0,0 +1,538 @@
+
+	// The reader does not refer to an MachO file.
+	if err != nil {
+		return fileError(filename, err)
+	}
+	var ni NativeImageMachO
+	ni.File = bi
+	return ni, nil
+}
+
+// newPE reads a Native Image from a Portable Executable file.
+func newPE(filename string, r io.ReaderAt) (NativeImage, error) {
+	// First attempt to read an ELF file.
+	bi, err := pe.NewFile(r)
+
+	// The reader does not refer to an MachO file.
+	if err != nil {
+		return fileError(filename, err)
+	}
+	var ni NativeImagePE
+	ni.File = bi
+	ni.Reader = r
+	return ni, nil
+}
+
+// FetchPkgs obtains the packages given in the binary.
+func (ni NativeImageElf) FetchPkgs() ([]pkg.Package, error) {
+	bi := ni.File
+	var sbom elf.Symbol
+	var sbomLength elf.Symbol
+	var svmVersion elf.Symbol
+
+	si, err := bi.Symbols()
+	if err != nil {
+		log.Debugf("native-image cataloger: no symbols found.")
+		return nil, err
+	}
+	for _, s := range si {
+		switch s.Name {
+		case nativeImageSbomSymbol:
+			sbom = s
+		case nativeImageSbomLengthSymbol:
+			sbomLength = s
+		case nativeImageSbomVersionSymbol:
+			svmVersion = s
+		}
+	}
+	if sbom.Value == 0 || sbomLength.Value == 0 || svmVersion.Value == 0 {
+		log.Debugf("native-image cataloger: %v", nativeImageMissingSymbolsError)
+		return nil, errors.New(nativeImageMissingSymbolsError)
+	}
+	dataSection := bi.Section(".data")
+	if dataSection == nil {
+		log.Debugf("native-image cataloger: .data section missing from ELF file.")
+		return nil, err
+	}
+	dataSectionBase := dataSection.SectionHeader.Addr
+	data, err := dataSection.Data()
+	if err != nil {
+		log.Debugf("native-image cataloger: cannot read the .data section.")
+		return nil, err
+	}
+	sbomLocation := sbom.Value - dataSectionBase
+	lengthLocation := sbomLength.Value - dataSectionBase
+
+	return decompressSbom(data, sbomLocation, lengthLocation)
+}
+
+// FetchPkgs obtains the packages from a Native Image given as a Mach O file.
+func (ni NativeImageMachO) FetchPkgs() ([]pkg.Package, error) {
+	var sbom macho.Symbol
+	var sbomLength macho.Symbol
+	var svmVersion macho.Symbol
+
+	bi := ni.File
+	for _, s := range bi.Symtab.Syms {
+		switch s.Name {
+		case "_" + nativeImageSbomSymbol:
+			sbom = s
+		case "_" + nativeImageSbomLengthSymbol:
+			sbomLength = s
+		case "_" + nativeImageSbomVersionSymbol:
+			svmVersion = s
+		}
+	}
+	if sbom.Value == 0 || sbomLength.Value == 0 || svmVersion.Value == 0 {
+		log.Debugf("native-image cataloger: %v.", nativeImageMissingSymbolsError)
+		return nil, errors.New(nativeImageMissingSymbolsError)
+	}
+
+	dataSegment := bi.Segment("__DATA")
+	if dataSegment == nil {
+		return nil, nil
+	}
+	databuf, err := dataSegment.Data()
+	if err != nil {
+		log.Debugf("native-image cataloger: cannot obtain buffer from data segment.")
+		return nil, nil
+	}
+	sbomLocation := sbom.Value - dataSegment.Addr
+	lengthLocation := sbomLength.Value - dataSegment.Addr
+
+	return decompressSbom(databuf, sbomLocation, lengthLocation)
+}
+
+type ExportTypesPE struct {
```

Comment:
similar export naming comment as above (could be not exported, lowercase names), although here the fields do not need to be exported, either. the same applies to nearly all exported names following, too